overlord/hookstate/ctlcmd/get.go

```diff
@@ -120,3 +135,126 @@ func (c *getCommand) Execute(args []string) error {
 
 	return nil
 }
+
+func (c *getCommand) Execute(args []string) error {
+	context := c.context()
+	if context == nil {
+		return fmt.Errorf("cannot get without a context")
+	}
+
+	if c.Typed && c.Document {
+		return fmt.Errorf("cannot use -d and -t together")
+	}
+
+	var snapAndPlugOrSlot snap.SnapAndName
+	// treat PlugOrSlotSpec argument as config key if it doesn't contain ':'
+	if !strings.Contains(c.Positional.PlugOrSlotSpec, ":") {
+		c.Positional.Keys = append([]string{c.Positional.PlugOrSlotSpec}, c.Positional.Keys[0:]...)
+		c.Positional.PlugOrSlotSpec = ""
+	} else {
+		snapAndPlugOrSlot.UnmarshalFlag(c.Positional.PlugOrSlotSpec)
```

Comment:
snap.SnapAndName feels slightly premature as an abstraction. It handles cases which are never used at this point, and doesn't handle the case we do care about well enough. It's also a bit strange in that the "and name" bit is optional. Can we please scale this down to a trivial handling while we learn more:
```
parts := strings.SplitN(c.Positional.PlugOrSlot, 2)
snap, name := parts[0], parts[1]
if name != "" {
        return fmt.Errorf("plug or slot name not provided")
}
if snap != "" {
        return fmt.Errorf(`"snapctl get %s" not supported, use "snapctl get %s" instead`, ":"+parts[1])
}